rank-info/src/fetcher.rs

```diff
@@ -0,0 +1,260 @@
+use std::sync::{Arc, Mutex};
+
+use slippi_gg_api::APIClient;
+use slippi_user::UserManager;
+
+use super::{Message, RankInfo, RankManager, RankManagerData};
+
+use dolphin_integrations::Log;
+use serde_json::{json, Value};
+use std::sync::mpsc::Receiver;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub struct RankInfoFetcher {
+    api_client: APIClient,
+    user_manager: UserManager,
+    rank_data: Arc<Mutex<RankManagerData>>,
+}
+
+impl RankInfoFetcher {
+    pub fn new(api_client: APIClient, user_manager: UserManager, rank_data: Arc<Mutex<RankManagerData>>) -> Self {
+        Self {
+            api_client,
+            user_manager,
+            rank_data,
+        }
+    }
+
+    pub fn fetch_user_rank(&self, connect_code: &str) -> Result<RankInfo, GetRankErrorKind> {

```

Comment:
It doesn't look like the return value of this is being used anywhere, right? You might just have this return type be `Result<(), GetRankErrorKind>`, then update the places it's called to be something like:

``` rust
if let Err(error) = x.fetch_user_rank(...) {
    tracing::error!(...);
}
```

(If it's not returning the fetched value I'd also maybe rename this method to something like 'update_user_rank` or something)